internal/shaderir/program.go

```diff
@@ -109,6 +111,7 @@ type Expr struct {
 	Swizzling   string
 	Index       int
 	Op          Op
+	Ast         ast.Expr
```

Comment:
`Expr` are stored by value. Is there a better way to search for a specific statement without pointers?